runtime/autoload/health.vim

```diff
@@ -78,6 +87,11 @@ function! s:indent_after_line1(s, columns) abort
   return join(lines, "\n")
 endfunction
 
+" Changes ':help clipboard' to '|clipoard|'. Also removes surrounding quotes.
+function! health#help_to_link(s) abort
```

Comment:
I'd even go as far as doing this indiscriminately on all input in `s:format_report_message()`. Then we can make this private (`s:help_to_link`) and avoid needing to think about it. That also makes it work for plugin authors.